### ISSUE TITLE
While deploying contract using mocks  getting this error  "Cannot read properties of undefined (reading 'length')"

### ISSUE THREAD

**Issue by Nikhil8400** [PatrickAlphaC/hardhat-fund-me-fcc#166]
i have used  await with get named accounts

**Comment by Nikhil8400** [comment]
//This is my deploy-mocks file

const {network, deployments} = require("hardhat")
const {developmentChains, DECIMALS, INITIAL_ANSWER} = require("../helper-hardhat-config.js")

module.exports = async ({ getNamedAccounts, deployment})=>{
    const {deploy, log} = deployments
    const{deployer} = await getNamedAccounts()
    const chainId = network.config.chainId
    
    
    if(chainId == 31337){
        log("Local network detected! Deploying mocks....")
        await deploy("MockV3Aggregator", {
            contract:"MockV3Aggregator",
            from: deployer,
            log:true,
            args:[DECIMALS,INITIAL_ANSWER]        //we have to pass constructor parameter for mockV3Aggregator
            //decimal parameter is going to equal to this decimal function...  and the _initial answer parameter tells with what does price feed starting with
            //Decimal should be placed 1st then place initial_answer accoding the constructor for MockV3Aggregator we can go and check in node modules files 
        })
        log("Mocks deployed!")
        log("___________________________________________________")
    }
}

module.exports.tags = ["all","mocks"]

**Comment by pirqqs** [comment]
Try wrapping your if statement with module.exports{} like this:
```javascript
module.exports = async ({ getNamedAccounts, deployments }) => {
    const { deploy, log } = deployments
    const { deployer } = await getNamedAccounts()

    if (developmentChains.includes(network.name)) {
        log("Local network detected! Deploying mocks...")
        await deploy("MockV3Aggregator", {
            contract: "MockV3Aggregator",
            from: deployer,
            log: true, //is deploying, what transaction is doing and where it is deployed with how much gas
            args: [DECIMALS, INITIAL_ANSWER],
        })
        log("Mocks deployed")
        log("-----------------------------------------------------------")
    }
    
    //We don't want to deploy this mock contract to a test net that actually has price feed
}
```

**Comment by pirqqs** [comment]
Try using this code
```javascript 
//Contract that we can use to deploy a fake price feed to a blockchain
const { network } = require("hardhat")
const { DEBUG_FILE_FORMAT_VERSION } = require("hardhat/internal/constants")
const { networkConfig } = require("../helper-hardhat-config")
const {
    developmentChains,
    DECIMALS,
    INITIAL_ANSWER,
} = require("../helper-hardhat-config")

module.exports = async ({ getNamedAccounts, deployments }) => {
    const { deploy, log } = deployments
    const { deployer } = await getNamedAccounts()

    if (developmentChains.includes(network.name)) {
        log("Local network detected! Deploying mocks...")
        await deploy("MockV3Aggregator", {
            contract: "MockV3Aggregator",
            from: deployer,
            log: true, //is deploying, what transaction is doing and where it is deployed with how much gas
            args: [DECIMALS, INITIAL_ANSWER],
        })
        log("Mocks deployed")
        log("-----------------------------------------------------------")
    }

    //We don't want to deploy this mock contract to a test net that actually has price feed
}

module.exports.tags = ["all", "mocks"]
```

**Comment by Hyratio** [comment]
Hello Nikhil8400, I don't know if it's still useful to you,but it might still be useful for someone else.. 
I had the same problem, (everytime there is a problem ahah) I solved it by looking at Patricks's hardhat.config.js in the github repo. Basically I had to add the deployer.1.  Before I had only  deployer: {default: 0}. This solved the problem to me. I hope it helps..

 namedAccounts: {
        deployer: {
            default: 0,
            1: 0,
        },
    },
}
